### ISSUE TITLE
Cant find new sensor 

### ISSUE THREAD

**Issue by winogregorio** [jonsag/pi-heating#2]
Thanks for turning this into a self-contained solution.  I had a few roadblocks along the way (requires stretch not latest buster, incorrect paths in a few instructions, etc) but with yours and JeffreyPowell's instructions, I was able to overcome them.  I am at the point where I scan for new sensors but the web page times out after about a minute with no new sensors found.  I edited the config file and the cat command of the device returns data.  I'll admit that I am pretty new to pi and it has been almost 2 decades since I've done anything close to this so maybe I am overlooking the obvious.  I hope you can provide the missing link.
Thanks,
Greg

**Comment by jonsag** [comment]
Been a long time since I did anything in this project.
Where/what was the wrong paths? Please tell me so I can correct it.
From my memory I recall something about an issue I had with timing out when scanning for devices. As I can recall it had something to do with incorrect subnet.
I'll look into it right now.

**Comment by jonsag** [comment]
Try running the following from cli:
`nmap -sP <your subnet> | grep report | grep -v router | rev | cut -d' ' -f 1 | rev | tr -d '()'`
\<your subnet\> could be for example 192.168.0.0/24
Does it find any clients?

Could you please explain your setup.
Are you running hub and remote, or combined hub/remote?

> I edited the config file and the cat command of the device returns data.

What config file and what cat command?

**Comment by winogregorio** [comment]
Thanks so much for taking the time to reply.  I tried the command below and it returned

PST

-Pn

seconds

 

I recall reading something about the subnet problem in the original instructions but could not find 192.168.x.x referenced in the sensors-list file.  This is where others corrected the problem.  My setup is a Pi Zero W running on a 10.222.0.0/24 network.

 

The trouble I had with the instructions were under Hook up 1-wire temp sensors to remote:

“Edit /home/pi/pi-heating-remote/configs/sensors and insert serials and names, for example 28-0516b4ff09ff = Out”  

should be 

“Edit /home/pi/pi-heating/pi-heating-remote/configs/sensors and insert serials and names, for example 28-0516b4ff09ff = Out”  

 

Also, there does not appear to be a web server running on 8081 yet the next lines refer to 8081 instead of 8080. When I use 8080 in the curl command I get a permission denied error.  When I try it from a emote web browser, I get file not found. 

 

Thanks again for looking at this,

Greg

**Comment by winogregorio** [comment]
I retried the nmap command because I just realized the output was not right.  It found all the devices on my 10.222.0.0/24 network.  However, my router is also connected to a 192.168.1.0 network via a VLAN.  I’m not sure that makes a difference.

**Comment by winogregorio** [comment]
I forgot to add, hub and remote on one device.  The config file I edited was the sensors file with the sensor serial number and name.  When I cat /sys/bus/w1/devices/28-00000b886971/w1_slave it returns

 

6e 01 4b 46 7f ff 02 10 71 : crc=71 YES

6e 01 4b 46 7f ff 02 10 71 t=22875

**Comment by jonsag** [comment]
Well, there's your problem.
Look into lines 169 and 171 in sensors-list.php to correct your problem.
Probably line 169 gets wrong subnet and continues to scan the wrong subnet and time outs.
One quick solution is to substitute '$subnet_cidr' on line 171 with '10.222.0.0/24' in your case.
What you want is the subnet of the pi.

There should actually be two servers running. The hub runs a server on port 8080 and the remote on 8081.
The file /etc/apache2/ports.conf should have two Listen-directives, 'Listen 8080' and 'Listen 8081', if you ran both pi-heating-hub-install.sh and pi-heating-remote-install.sh on the same machine.
If you correct the file manually, remember to restart apache afterwards.

I don't have this up and running anywhere, so I can't confirm any of this right now.

About the paths in the README,
actually the path for the 'sensors'-file should be as it says, as per line 118 in pi-heating-remote-install.sh.
The path to the 'sensors'-file referred to in the README will be there _after_ the install.

**Comment by winogregorio** [comment]
SUCCESS!!! Thank you so much, again.  I ran pi-heating-remote-install.sh and the webserver is now running on 8081 as well as 8080.  The instructions were not clear on that.  In fact, I don't think they mentioned it.  
I did not have to edit the sensors-list.php file.  It appears to get the subnet information from the interface.  Now all I need to do is figure out how to output in Fahrenheit instead of Celsius I can get to work building the rest of them
Thanks,
Greg

**Comment by jonsag** [comment]
Great.
Why not start a little revolution and join the rest of the world using the more sensible celsius scale! :-)

**Comment by winogregorio** [comment]
> 
> 
> Great.
> Why not start a little revolution and join the rest of the world using the more sensible celsius scale! :-)

I'm fully on board with that.  It's the wife I'm trying to please LOL!

**Comment by jonsag** [comment]
Oh.
I know how important that is...
Good luck with everything
/jon

**Comment by jonsag** [comment]
Come to think of it now, perhaps the easiest way is to start with the file 'value.php' where in line 29 the value is calculated.
Add a line below that with something like:

`$value = $value * 9/5 + 32;`

and then change every occurence of 'C' to 'F' in the web GUI

**Comment by winogregorio** [comment]
Thanks!  I was working backward to that. I edited line 29 to add *(9/5)+32 at the end.  Seems to have worked to convert the output to F.  

 

However, in an effort to delete the historical data which still had the C values, I deleted the sensor and added it back which increased the Sensor ID value by 1.  To test it, I deleted and added it back a second time and it increased again by 1.  It now reads 3.  Probably in a config file someplace.  Overall, this is a minor detail that seems to have no impact.  I probably will not bother to investigate further.

 

I can’t thank you enough for your time.

Greg
